src/main/java/com/entropyteam/entropay/employees/services/PtoService.java

```diff
@@ -192,4 +193,20 @@ private CalendarEventDto createCalendarEventDto(Pto pto){
         String eventName = employee.getFirstName() + " " + employee.getLastName() + " " + leaveType.getName();
         return new CalendarEventDto(eventId, eventName, startDate, endDate);
     }
+
+    public PtoDto cancelPto(UUID id, PtoDto ptoDto) {
+        Pto oldEntity = ptoRepository.findById(id).orElseThrow();
+        Pto entityToUpdate = toEntity(ptoDto);
+        entityToUpdate.setId(id);
+        entityToUpdate.setStatus(entityToUpdate.getStatus());
+        if (isVacationType(oldEntity) && isVacationType(entityToUpdate) && oldEntity.getStatus().equals(Status.APPROVED)
+                && ptoDto.status().equals(Status.CANCELLED)) {
+            vacationService.addVacationCredit(entityToUpdate.getEmployee(), entityToUpdate.getDaysAsInteger());
+            googleService.deleteGoogleCalendarEvent(id.toString());
```

Comment:
in all cases we need to delete the google calendar event

Reply:
Resolved